### PR TITLE
Launch v1 client updates

### PR DESCRIPTION
**Pull request by phil-scale**
Updates the Launch python client to use v1 endpoints when available. Deprecated methods will be deleted as part of the major version change. The missing routes in v1 are currently bundle upload and batch inference.

The plan is to keep these changes in the feature branch `launch-v1`. When the migration is fully ready, we can then merge `launch-v1` into `master` and publish to PyPI.

[[sc-637813]](https://app.shortcut.com/scaleai/story/637813/update-launch-client-to-use-v1-routes)

### CODE REVIEW

File: launch/client.py [hunk 1]
```diff
@@ -630,36 +661,39 @@ def create_model_endpoint(
...
-                min_workers=min_workers,
-                max_workers=max_workers,
-                per_worker=per_worker,
-                endpoint_type=endpoint_type,
-                post_inference_hooks=post_inference_hooks,
-                labels=labels or {},
-            )
-            if gpus == 0:
-                del payload["gpu_type"]
-            elif gpus > 0 and gpu_type is None:
```
Comment (seanshi-scale) [inline launch/client.py:651]: looks like we're removing a bit of client-side validation but that's probably not a big deal

File: launch/model_endpoint.py [hunk 2]
```diff
@@ -30,6 +30,11 @@ class ModelEndpoint:
     The name of the endpoint. Must be unique across all endpoints owned by the user.
     """
 
+    id: Optional[str] = None
+    """
+    A globally unique identifier for the bundle.
```
Comment (Tianwei-She) [inline launch/model_endpoint.py:35]: for the endpoint?

**Review by Tianwei-She — APPROVED** [review]

File: launch/client.py [hunk 3]
```diff
@@ -728,35 +762,54 @@ def edit_model_endpoint(
...
+
+            if model_bundle is None:
+                model_bundle_id = None
+            elif (
+                isinstance(model_bundle, ModelBundle)
+                and model_bundle.id is not None
+            ):
+                model_bundle_id = model_bundle.id
+            else:
+                model_bundle = self.get_model_bundle(model_bundle)
```
Comment (seanshi-scale) [inline launch/client.py:780]: is this to translate from model bundle names to ids?

**Review by seanshi-scale — APPROVED** [review]

File: launch/model_endpoint.py (hunk 2 above)
Comment (phil-scale) [reply launch/model_endpoint.py:35]: yeah thanks for catching!

File: launch/client.py (hunk 1 above)
Comment (phil-scale) [reply launch/client.py:651]: yeah oops - though I think there's server-side validation for this now

**Pull request merged**